examples/dataflow-hashpipeline/hashpipeline/buffering.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# pytype: skip-file
+
+from __future__ import absolute_import
+from apache_beam.transforms.userstate import BagStateSpec, TimerSpec, CombiningValueStateSpec, on_timer
+from apache_beam.transforms.timeutil import TimeDomain
+from apache_beam.coders import VarIntCoder, TupleCoder, StrUtf8Coder
+from apache_beam.transforms.combiners import CountCombineFn
+import apache_beam as beam
+import time
+from abc import ABC, abstractmethod
+from collections import defaultdict
+
+class BufferedPayloadFactory(ABC):
+  def __init__(self, max_size):
+    self.payloads = []
+    self.max_size = max_size
+
+  @abstractmethod
+  def running_payload(self):
+    '''Returns a singleton instance of the current payload'''
+    raise NotImplementedError
```

Comment:
nit: abc decorator will raise this for you.
```suggestion
    pass
```